controllers/expenseController.js

```diff
@@ -0,0 +1,113 @@
+const Expense = require('../models/Expense');
+const expensesSchema = require('../models/expensesJoi'); // Importuj schemat walidacji
+
+// Funkcja dodawania wydatków
+const addExpense = async (req, res) => {
+  try {
+    // Walidacja danych wejściowych
+    const { error } = expensesSchema.validate(req.body);
+    if (error) {
+      return res.status(400).json({ message: error.details[0].message });
+    }
+
+    const { description, amount, date, category } = req.body;
+
+    const newExpense = new Expense({
+      description,
+      amount,
+      date,
+      category
+    });
+
+    await newExpense.save();
+
+    // Miejsce na logikę aktualizacji salda użytkownika itp.
+
+    res.status(200).json({
+      message: 'Expense added successfully',
+      expense: newExpense
+    });
+  } catch (error) {
+    console.error('Error adding expense:', error);
+    res.status(500).json({ message: 'Internal server error' });
+  }
+};
+
+// Funkcja pobierania wydatków
+const getExpenses = async (req, res) => {
+  try {
```

Comment:
ale get oddaje pustą tabelę i listę miesięcy z 0, najprawdopodobniej przez to że prubuje się odnieść do właściciela danego expensa a to trzeba stworzyć